### DISCUSSION TITLE
Add missing major Gerrit features

### DISCUSSION THREAD

**Discussion by willfaught** [community/community#18879] [Pull Requests]
My understanding is that those are:

- The ability to review individual commits, including their messages (and possibly other things of interest, like security/signing data/metadata, or the author/committer, or the time).
- The ability to require a minimum number of reviews/approvals, and the ability to give certain users "+2" powers to immediately approve PRs.

These seem to be the major sticking points for the https://github.com/golang/go repo (one of the most-starred repos), and why they still use Gerrit for reviews, and it causes a large amount of friction for contributing to that project. They wrote a whole bot to mirror the basics of GitHub PRs as Gerrit submissions, but it doesn't propagate comments back and forth, etc. It's hardly nice, and I doubt GitHub likes its users being sent off-site.

Personally, I'd love to have the ability to review commits individually, to comment on their messages, and to comment on whether a commit should be split into multiple, or whether multiple commits should be combined into one, etc. Help us keep our commit histories clean and useful. *Please* add this!

**Comment by rambleraptor** [comment]
Stacked diffs would also be wonderful!

**Comment by yoav85** [comment]
Great suggestion!

Also, as a reviewer - I'd like to:
1. See all changes since my last review of this PR (already implemented) while ignoring changes which are a result of merging (or rebasing) the PR's target branch into the PR's working branch.
2. When re-reviewing a PR, as part of the "files changed" view - show **all** of the unresolved comments, even if they were about lines which have been moved or deleted since then. The current UI forces the reviewer to check both the "files changed" tab and the discussion tab, and manually "sync" between them.

**Comment by kreintjes** [comment]
It would also be great if the ability to review individual commits would also make it possible to add code suggestions (` ```suggestion`) per commit. This should be fairly easy for lines that don't change in a later commit, although currently not possible (or at least no button for it). It would be even nicer if it is possible for lines that change again later as well (were the diff shows the old version like it was directly before the commit being reviewed).

UPDATE: apparently there are also separate feature requests for that already https://github.com/github-community/community/discussions/7128 / https://github.com/github-community/community/discussions/15626
